packages/blob/src/multipart/upload.ts

```diff
@@ -174,6 +173,27 @@ export function uploadAllParts({
     let arrayBuffers: ArrayBuffer[] = [];
     let currentPartBytesRead = 0;
 
+    let onUploadProgress: (() => void) | undefined;
+    const totalLoadedPerPartNumber: Record<string, number> = {};
+
+    if (options.onUploadProgress) {
+      onUploadProgress = throttle(() => {
```

Comment:
why do we throttle this? Doesn't this make the upload feel slower

Reply:
I chose to throttle at 100ms as it looked like most libraries providing uploadProgress were also throttling sending events (example: https://github.com/axios/axios/blob/v1.x/lib/helpers/progressEventReducer.js).

I guess it's here to provide a better experience in most cases because otherwise, you can receive a LOT of progress events.

With throttling:

https://github.com/user-attachments/assets/49435109-67ec-4158-81ed-bca20520d40a

Without throttling:

https://github.com/user-attachments/assets/9ef3c97a-9146-45b6-949d-ebbcf42131d2

I am happy to change my mind around this if we think this could be bad for customers, it's also something we can just remove without breaking the API if needed (rare case!)

Reply:
I see. First video looks way nicer 👍